website/demos/CommonFeatures.tsx

```diff
@@ -149,28 +150,49 @@ function getColumns(countries: string[]): readonly Column<Row, SummaryRow>[] {
           </>
         );
       },
-      editor({ row, onRowChange, onClose }) {
+      editor({ row, column, onRowChange, onClose }) {
         return (
-          <div className={dialogContainerClassname}>
-            <dialog open>
-              <input
-                autoFocus
-                type="range"
-                min="0"
-                max="100"
-                value={row.progress}
-                onChange={(e) => onRowChange({ ...row, progress: e.target.valueAsNumber })}
+          <>
+            {createPortal(
+              <div
+                className={dialogContainerClassname}
+                onKeyDown={(event) => {
+                  if (event.key === 'Escape') {
+                    onClose();
+                  }
+                }}
+              >
+                <dialog open>
+                  <input
+                    autoFocus
+                    type="range"
+                    min="0"
+                    max="100"
+                    value={row.progress}
+                    onChange={(e) => onRowChange({ ...row, progress: e.target.valueAsNumber })}
+                  />
+                  <menu>
+                    <button onClick={() => onClose()}>Cancel</button>
+                    <button onClick={() => onClose(true)}>Save</button>
+                  </menu>
+                </dialog>
+              </div>,
+              document.body
+            )}
+            <div
+              className={css`
+                padding: 0 8px;
```

Comment:
This might be bothersome if we have to do it many times 🤔

Reply:
Yeah, may be we can return formattedValue with editorProps or formatterClassname or we can set a flag in editorProps like isPortal or something and render the formatter also.